syntax = "proto3";

service SpeechService {
  // ìŠ¤íŠ¸ë¦¼ìœ¼ë¡œ ë°ì´í„°ë¥¼ ì˜ê² ë‹¤ ì„ ì–¸
  rpc SendAudioStream (stream AudioRequest) returns (AudioResponse);
}

// ğŸ“¦ ë°ì´í„° ë°•ìŠ¤ ëª¨ì–‘ ì •ì˜
message AudioRequest {
  bytes audio_data = 1;  // ì‹¤ì œ ì˜¤ë””ì˜¤ (ì´ì§„ìˆ˜)
  bool is_final = 2;     // "ë§ ëë‚¬ì–´?" (True/False)
}

message AudioResponse {
  string text = 1;       // AI ë‹µë³€ (STT + Chat ê²°ê³¼)
}
